keps/sig-node/2727-grpc-probe/README.md

```diff
@@ -0,0 +1,447 @@
+  //
+  // The service name can be the empty string (i.e. "").
+  Service string `json:"service" protobuf:"bytes,2,opt,name=service"`
+
+  // Host is the host name to connect to, defaults to the Pod's IP.
+  Host string `json,omitempty", protobuf:"bytes,3,opt,name=host"`
+}
+```
+
+### Test Plan
+
+- Unit test: Add unit tests to `pkg/kubelet/prober/...`
+- e2e: Add test case and conformance test to `e2e/common/node/container_probe.go`.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Implement the feature.
+- Add unit and e2e tests for the feature.
+
+#### Beta
+
+- Solicit feedback from the Alpha. Validate that API is appropriate
+  for users. There are some potential tunables:
+  - `User-Agent`
+  - connect timeout
+  - protocol (HTTP, QUIC)
+- Ensure tests are stable and passing.
+
+Depending on skew strategy:
+
+- kubelet version skew ensures all (kubelet ver, cluster ver) support
+  the feature.
+
+#### GA
+
+- Address feedback from beta
+- Close on any remaining open issues & bugs
+
+### Upgrade / Downgrade Strategy
+
+Upgrade: N/A
+
+Downgrade: gRPC probes will not be supported in a downgrade from Alpha.
+
+### Version Skew Strategy
+
+- We may not be able to graduate this widely until all kubelet version
+  skew supports the probe type.
+
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+Feature enablement will be guarded by a feature gate flag.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `GRPCContainerProbe`
+  - Components depending on the feature gate: `kubelet`
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It becomes enabled.
```

Comment:
To re-enable it, you are restarting kubelet, so the grpc probes would start running again.